code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,315 @@
 	owner.heal_limb_damage(0, 5)
 	return ..()
 
+// ***************************************
+// *********** Essence Link
+// ***************************************
+/obj/effect/ebeam/essence_link
+	name = "essence link beam"
+
+/datum/status_effect/stacking/essence_link
+	id = "xeno_essence_link"
+	tick_interval = 2 SECONDS
+	stacks = 0
+	stack_decay = -1 //Not meant to decay.
+	max_stacks = 3
+	consumed_on_threshold = FALSE
+	alert_type = null
+	/// The owner of the link.
+	var/mob/living/carbon/xenomorph/link_owner
+	/// Whom the owner is linked to.
+	var/mob/living/carbon/xenomorph/link_target
+	/// References the Essence Link action and its vars.
+	var/datum/action/xeno_action/activable/essence_link/essence_link_action
+	/// If the target xeno was within range.
+	var/was_within_range = TRUE
+	/// Cooldown for passive attunement increase.
+	COOLDOWN_DECLARE(attunement_increase)
+	/// Delay between plasma warnings. Plasma warnings are issued if there's not enough plasma for the passive regeneration.
+	var/plasma_warning_cooldown = 20 SECONDS
+	/// Cooldown for the plasma warning.
+	COOLDOWN_DECLARE(plasma_warning)
+	/// The beam used to represent the link between linked xenos.
+	var/datum/beam/current_beam
+	/// Used to prevent duplicate shared heals.
+	var/shared_heal_count
+
+/datum/status_effect/stacking/essence_link/on_creation(mob/living/new_owner, stacks_to_apply, mob/living/carbon/link_target)
+	link_owner = new_owner
+	src.link_target = link_target
+	essence_link_action = link_owner.actions_by_path[/datum/action/xeno_action/activable/essence_link]
+	ADD_TRAIT(link_owner, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(link_target, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	RegisterSignal(link_owner, COMSIG_MOB_DEATH, .proc/handle_death)
+	RegisterSignal(link_target, COMSIG_MOB_DEATH, .proc/handle_death)
+	RegisterSignal(link_owner, COMSIG_MOVABLE_MOVED, .proc/handle_dist)
+	RegisterSignal(link_target, COMSIG_MOVABLE_MOVED, .proc/handle_dist)
```

Comment:
I'm concerned about performance from calling this every time either linkee moves. How much would you lose by checking it during tick instead?

Reply:
Not much. Tick would be a bit too overloaded, and the link might be a little bit inaccurate when updating due to tick intervals, but it's otherwise not that much of a loss or anything.

Reply:
Consulted Bravemole, they think it's a good idea. Doing the change.

Reply:
Done.